slack-plugin/src/main/kotlin/slack/gradle/SlackExtension.kt

```diff
@@ -636,12 +645,73 @@ public abstract class DaggerHandler @Inject constructor(objects: ObjectFactory)
 
 @SlackExtensionMarker
 @Suppress("UnnecessaryAbstractClass")
-public abstract class AndroidHandler @Inject constructor(objects: ObjectFactory) {
+public abstract class ComposeHandler
+@Inject
+constructor(
+  objects: ObjectFactory,
+  globalSlackProperties: SlackProperties,
+  slackProperties: SlackProperties,
+  versionCatalog: VersionCatalog
+) {
+
+  private val composeBundleAlias =
+    globalSlackProperties.defaultComposeAndroidBundleAlias?.let { alias ->
+      versionCatalog.findBundle(alias).orElse(null)
+    }
+  private val composeCompilerVersion by lazy {
+    slackProperties.versions.composeCompiler
+      ?: error("Missing `compose-compiler` version in catalog")
+  }
+  internal val enabled = objects.property<Boolean>().convention(false)
+
+  /**
+   * This is weird! Due to the non-property nature of the AGP buildFeatures and composeOptions DSLs,
+   * we can't lazily chain their values to our own extension's properties. Because of this, we
+   * lazily set this instance from [StandardProjectConfigurations] during Android extension
+   * evaluation and then make calls to [enable] directly set the values on this instance. Ideally we
+   * could eventually remove this if/when AGP finally makes these properties lazy.
+   */
+  internal var androidExtension: CommonExtension<*, *, *, *>? = null
+
+  internal fun enable() {
+    val extension =
+      checkNotNull(androidExtension) {
+        "ComposeHandler must be configured with an Android extension before it can be enabled. Did you apply the Android gradle plugin?"
+      }
+    enabled.set(true)
```

Comment:
One thing I like to do after calling this is to then call `enabled.disallowChanges()`. That way, I know that the value can only ever be set once. Probably less relevant in this case, since it can only be "enabled", but we have a lot of true|false settings.

Reply:
Unrelated, and this is a matter of taste more than anything I think, but in some cases I've delegated the call to a companion object on a specific plugin, solely so that I could "encapsulate" all related logic in one place. E.g., we have a "UnitTestPlugin", and so we'll call static methods on that from a Handler class like this, again just to keep all test-related logic in one place for discoverability.

Reply:
Could you unpack that a bit more for me?

Reply:
disallow changes here [9d5a5c9](https://github.com/slackhq/slack-gradle-plugin/pull/108/commits/9d5a5c9bae9c4eb4464d1506a5eea6815bd6f9aa). Probably more places we should be doing this tbh